### PR TITLE
Changed Swagger Normalizer to correctly support nested serialization …

### PR DESCRIPTION
**Pull request by jonadoe**
…groups

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #986 #988 #940 
| License       | MIT
| Doc PR        | not needed

Fixed the bug from Issue #986 and also fixed general handling of serialization groups in swagger normalizer.

### CODE REVIEW

File: src/Swagger/Serializer/DocumentationNormalizer.php [hunk 1]
```diff
@@ -49,6 +49,10 @@
     private $filterCollection;
     private $nameConverter;
 
+    private $mimeTypes;
```
Comment (dunglas) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: Can you get rid of those states please? A service should not have states like those ones.

File: src/Swagger/Serializer/DocumentationNormalizer.php [hunk 2]
```diff
@@ -67,36 +71,33 @@ public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFa
      */
     public function normalize($object, $format = null, array $context = [])
```
Comment (dunglas) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: All changes to this method must be reverted.

File: src/Swagger/Serializer/DocumentationNormalizer.php (hunk 1 above)
Comment (jonadoe) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: The variables get reset on every call of the normalize function, so they are not really states.
The only other way to solve this is to pass through these variables (at least the $definitions) to nearly all other methods in the call. This makes the code much harder to read and does not make any sense at all.
If you like I can implement it that way, but I think it's better like it is now

Comment (dunglas) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: I would prefer to pass those variables as arguments like it was before.

File: src/Swagger/Serializer/DocumentationNormalizer.php (hunk 2 above)
Comment (dunglas) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: WDYT @api-platform/core-team?

File: src/Swagger/Serializer/DocumentationNormalizer.php (hunk 1 above)
Comment (jonadoe) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: Ok, I think this will make the code more complicated, but I will change it.

**Comment by jonadoe** [comment]
@dunglas 
The requested changes are now implemented

**Comment by jonadoe** [comment]
I'm currently short on time, but I will add a test and rebase the branch soon.

**Comment by dunglas** [comment]
@jonadoe do you think that you'll be able to add a test soon? I plan to release a new bug fix version in the beginning of the next week.

File: src/Swagger/Serializer/DocumentationNormalizer.php [hunk 3]
```diff
@@ -333,25 +341,24 @@ private function updateDeleteOperation(\ArrayObject $pathOperation, string $reso
...
         if (isset($serializerContext['groups'])) {
             $definitionKey = sprintf('%s_%s', $resourceMetadata->getShortName(), md5(serialize($serializerContext['groups'])));
         } else {
             $definitionKey = $resourceMetadata->getShortName();
         }
-        $definitions[$definitionKey] = $this->getDefinitionSchema($resourceClass, $resourceMetadata, $serializerContext, $definitionKey);
+
+        if (!isset($definitions[$definitionKey])) {
+            $definitions[$definitionKey] = [];  // Initialize first to prevent infinite loop
+            $definitions[$definitionKey] = $this->getDefinitionSchema($resourceClass, $resourceMetadata, $definitions, $serializerContext);
```
Comment (soyuka) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: This shouldn't be outside of the condition?

**Comment by jonadoe** [comment]
@dunglas I will be on vacation for one week and can't do it before 24/25th, sorry

**Comment by soyuka** [comment]
I'll add the test asap.

**Comment by jonadoe** [comment]
Thank you @soyuka !!

**Comment by dunglas** [comment]
Thank you very much @jonadoe and @soyuka as always :)

**Pull request merged**